pytorch_to_returnn/converter/converter.py

```diff
@@ -156,6 +157,26 @@ def _make_tf_feed_dict(self, input: Data):
     else:
       for i, size in input.size_placeholder.items():
         d[size] = [self._inputs_np.shape[input.get_batch_axis(i)]] * n_batch  # not so relevant
+    if out_entry:
+      non_deterministic_tensor_entries =  []
+      entries_to_visit = [out_entry]
+      visited_entries = set()
+      while entries_to_visit:
+        entry_ = entries_to_visit.pop(0)
+        if entry_ in visited_entries:
+          continue
+        visited_entries.add(entry_)
+        if not all(call_.module.module.is_deterministic for call_ in entry_.output_from_calls):
+          non_deterministic_tensor_entries.append(entry_)
+        for call_ in entry_.output_from_calls:
+          for prev_entry_ in [pe for pe in call_.inputs_args if isinstance(pe, TensorEntry)]:
```

Comment:
I would not create this temp list here. You can simply move the check `if isinstance(prev_entry_, TensorEntry)` into the loop.

But also, I think you should not loop over `input_args`. That can be arbitrarily nested. But even when you flatten this, it could still miss things (also e.g. from kwargs). The right way would be to iterate over `inputs_tensor_deps`.